framework/include/parser/CommandLine.h

```diff
@@ -191,3 +191,13 @@ CommandLine::search(const std::string & option_name, T & argument)
   }
   mooseError("Unrecognized option name");
 }
+
+enum HitCmpResult
```

Comment:
You might look at `enum class` instead of `enum`. The former is the newer way that gives you more type safety.